test/gtest/common/mem_buffer.cc

```diff
@@ -268,7 +274,12 @@ void mem_buffer::release(void *ptr, ucs_memory_type_t mem_type)
 #if HAVE_CUDA
         case UCS_MEMORY_TYPE_CUDA:
         case UCS_MEMORY_TYPE_CUDA_MANAGED:
-            CUDA_CALL(cudaFree(ptr), ": ptr=" << ptr);
+            if (!async) {
```

Comment:
async branch is not applicable to CUDA managed memory?

Reply:
Removed branch from CUDA_MANAGED case.